rocksdb_replicator/replicated_db.cpp

```diff
@@ -139,11 +146,32 @@ rocksdb::Status RocksDBReplicator::ReplicatedDB::Write(
     // TODO(bol): This potentially could block all worker threads. We may
     // consider having a dedicated set of worker threads for admin requests,
     // and/or provide async write API when this turns out to be a problem.
-    if (!max_seq_no_acked_.wait(cur_seq_no, FLAGS_replicator_timeout_ms)) {
+    if (!max_seq_no_acked_.wait(cur_seq_no, current_replication_timeout_ms_)) {
       incCounter(kReplicatorWriteWaitTimedOut, 1, db_name_);
       LOG(ERROR) << "Failed to receive ack from follower, timing out for " << db_name_;
+      numConsecutiveReplTimeout_ ++;
```

Comment:
Do these operations need to be thread safe? Multiple threads could be writing to the same DB simultaneously, right?